app/src/main/java/co/nimblehq/ui/base/BaseViewModel.kt

```diff
@@ -32,5 +36,46 @@ abstract class BaseViewModel : ViewModel() {
     val navigator: Observable<NavigationEvent>
         get() = _navigator
 
+    /**
+     * To show loading manually, should call `hideLoading` after
+     */
+    protected fun showLoading() {
+        if (loadingCount == 0) {
+            _showLoading.onNext(true)
+        }
+        loadingCount++
+    }
+
+    /**
+     * To hide loading manually, should be called after `showLoading`
+     */
+    protected fun hideLoading() {
```

Comment:
hmm, basically yes, but I tend to keep them as `protected` for manual usages in case we don't like to use `doShowLoading()` operator 💭  One use case for manual usages is: webview loading between page navigations. 🤓